pkg/steps/source.go

```diff
@@ -591,3 +609,17 @@ func trimLabels(labels map[string]string) map[string]string {
 	}
 	return labels
 }
+
+func getSourceSecretFromName(secretName string) *coreapi.LocalObjectReference {
```

Comment:
I don't really see the utility in having these as functions

Reply:
this is a helper function to check to avoid panics when accessing the `secret.Name` when secret is nil.

Reply:
I don't want to pass empty LocalObjectReference, otherwise, I will have to update all the integration tests adding the empty object. I would prefer having this helper function to pass nil when there is no secret

Reply:
You only use it in one place, though, so why not just do the `if` logic there? That's what I mean.

Reply:
I am using it in `source` `git_source` and `rpm_injection`. That's why I made this a function.  However, If you think that it's ok, I could put the `if` logic to those places.

Reply:
No, if we're reusing then we should keep it